internal/cluster/distribution/eks/eksprovider/workflow/amazon.go

```diff
@@ -483,15 +483,28 @@ func WaitForASGToBeFulfilled(
 					if autoscaling.IsErrorFinal(err) {
 						return errors.WithDetails(err, "nodePoolName", nodePoolName, "stackName", aws.StringValue(asGroup.AutoScalingGroupName))
 					}
-					// log.Debug(err)
 					continue
 				}
 				if ok {
-					// log.Debug("ASG is healthy")
+					logger.Debug("ASG is healthy")
 					return nil
 				}
 			} else {
-				return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s", asgFulfillmentWaitAttempts, asgFulfillmentWaitInterval)
+				asGroup, err := m.GetAutoscalingGroupByStackName(stackName)
+				if err != nil {
+					return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s, could not get ASG in stack: %s", asgFulfillmentWaitAttempts, asgFulfillmentWaitInterval, stackName)
+				}
+				lastActivity, _ := m.LastAutoscalingActivity(ctx, asGroup.AutoScalingGroupName)
+				if lastActivity != nil {
+					return errors.Errorf(
+						"waiting for ASG to be fulfilled timed out after %d x %s, status: %s, cause: %s, description: %s",
+						asgFulfillmentWaitAttempts,
+						asgFulfillmentWaitInterval,
+						aws.StringValue(lastActivity.StatusCode),
+						aws.StringValue(lastActivity.Cause),
+						aws.StringValue(lastActivity.Description),
+					)
+				}
```

Comment:
**Question**: shouldn't we do the same here as [there](https://github.com/banzaicloud/pipeline/pull/3591/files#diff-3ee9d719848b4882a759b0611d32bbf3df439588b845ba3f528f1819c5c4b271R89-R101)?

Reply:
Again, thanks for noticing! 🙂